dcrMetric.py

```diff
@@ -0,0 +1,301 @@
+        ----------
+        kde: `scipy.stats.kde.gaussian_kde`, optional
+            Uses ``self.kde`` if not supplied.
+        
+        Returns
+        -------
+        metric: `float`
+            The metric evaluating how well DCR is constrained for the given KDE.
+        """
+        if kde is None:
+            self.checkKde()
+            kde = self.kde
+        metric = self._calculateMetric(self.density, self.kde.dataset)
+        return metric
+    
+    def _calculateMetric(self, density_kde=None, visit_measures=None, use_reference=True):
+        if density_kde is None:
+            density_kde = self.density
+        if visit_measures is None:
+            visit_measures = self.kde.dataset
+        visit_weights = 1./(density_kde(visit_measures))**2
+        metric = np.sqrt(np.sum(visit_weights))
+        if use_reference:
+            metric /= self.metric_reference
+        return metric
+    
+    def evaluateKde(self, visit_measure):
+        """Evaluate how well DCR is constrained for an observation.
+        
+        Parameters
+        ----------
+        visit_measure: `float`
+            The one-dimensional parameterization of airmass and hour angle.
+        
+        Returns
+        -------
+        visit_metric: `float`
+            The metric evaluating how well a new observation can be constrained for the current DCR model.
+        """
+        template_metric = self.metric
+        visit_metric = self.kde(visit_measure)
+        visit_metric *= self.norm*template_metric
+        return visit_metric
+    
+    def evaluateVisit(self, airmass, hour_angle):
+        """Evaluate how well DCR is constrained for an observation.
+        
+        Parameters
+        ----------
+        airmass: `float`
+            The observed airmass of the observation.
+        hour_angle: `float`
+            The observed hour angle of the observation.
+        
+        Returns
+        -------
+        visit_metric: `float`
+            The metric evaluating how well a new observation can be constrained for the current DCR model.
+        """
+        self.checkKde()
+        visit_measure = self.parameterizeVisit(airmass, hour_angle)
+        visit_metric = self.evaluateKde(visit_measure)
+        return visit_metric
+        
+    def evaluateVisitInfo(self, visitInfo):
+        """Evaluate how well DCR is constrained for an observation.
+        
+        Parameters
+        ----------
+        visitInfo: `lsst.afw.image.VisitInfo`
+            The VisitInfo metadata of an exposure.
+        
+        Returns
+        -------
+        visit_metric: `float`
+            The metric evaluating how well a new observation can be constrained for the current DCR model.
+        """
+        hour_angle = visitInfo.getBoresightHourAngle().asRadians()
+        airmass = visitInfo.getBoresightAirmass()
+        return self.evaluateVisit(airmass, hour_angle)
+    
+    def plotMetric(self, window=1000, plot_measures=True):
```

Comment:
Please give a little more detail in the docstring about what this is plotting. I stared at these plots for a while in the notebook until I worked out that a larger DCR metric is what you're aiming for and the x-axis is your visit measure parameter.